src/cmake/AxomMacros.cmake

```
@@ -115,6 +115,96 @@ macro(axom_add_component)
 endmacro(axom_add_component)
 
 
+##------------------------------------------------------------------------------
+## axom_add_executable(
+##                     NAME        <name>
+##                     SOURCES     [source1 [source2 ...]]
+##                     HEADERS     [header1 [header2 ...]]
+##                     INCLUDES    [dir1 [dir2 ...]]
+##                     DEFINES     [define1 [define2 ...]]
+##                     DEPENDS_ON  [dep1 [dep2 ...]]
+##                     OUTPUT_DIR  [dir]
+##                     OUTPUT_NAME [name]
+##                     FOLDER      [name])
+##
+## Wrapper around blt_add_executable
+##------------------------------------------------------------------------------
+macro(axom_add_executable)
+
+    set(options )
+    set(singleValueArgs NAME OUTPUT_DIR OUTPUT_NAME FOLDER)
+    set(multiValueArgs HEADERS SOURCES INCLUDES DEFINES DEPENDS_ON)
+
+    # Parse the arguments to the macro
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    # Blanket add openmp as a dependency to get around not propagating
+    # openmp to fortran dependencies
+    blt_list_append(TO arg_DEPENDS_ON ELEMENTS openmp IF ENABLE_OPENMP)
```

I guess (?)

Should this be
```suggestion
    # Blanket add openmp as a dependency to get around not propagating
    # openmp to fortran dependencies
    blt_list_append(TO arg_DEPENDS_ON ELEMENTS openmp IF AXOM_ENABLE_OPENMP)
```
e.g. in case the BLT variable is set, but the axom one is not? Or would we want it in both cases?
(we've had so many issues/bugs related to OpenMP...)

---

Ah yes.. i copied this from the previous attempt at this. Good catch.